hbase_sql/src/main/java/com/splicemachine/stream/MessageBackupHandler.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2012 - 2021 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.stream;
+
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Queue;
+import java.util.UUID;
+import java.util.concurrent.*;
+
+/**
+ * If the cache mode is enabled, it starts one thread to check the timeout on message consumption and
+ * if the timeout occurs, the client does not consume messages, it starts to cache messages from Spark to the file
+ * system.
+ */
+public class MessageBackupHandler {
+    private static final Logger LOG = Logger.getLogger(MessageBackupHandler.class);
+
+    static final int CONSUMPTION_TIMEOUT = 300000;
+
+    private UUID uuid;
+    private HdfsStreamWriter hdfsStreamWriter;
+    private ScheduledExecutorService timeoutMonitorExecutor;
+    private ExecutorService backupExecutor;
+
```

Comment:
try to use general purpose thread pool from SIDriver if possible.

Reply:
backupExecutor is replaced
timeoutMonitorExecutor is still initialized here, as there is no general ScheduledExecutorService